pkg/api/auth.go

```diff
@@ -0,0 +1,190 @@
+	}
+	globalPass, ok := os.LookupEnv("TODO_PASSWORD")
+	if !ok || len(globalPass) == 0 {
+		globalPass = ""
+	}
+
+	body := req.Body
+	defer body.Close()
+
+	err := json.NewDecoder(body).Decode(&creds)
+	if err != nil {
+		errText := "ошибка десериализации JSON"
+		logger.Printf("%s: %v", errText, err)
+		err = jsonError(res, errText, http.StatusBadRequest)
+		if err != nil {
+			logger.Println(err)
+		}
+		return
+	}
+
+	if creds.Password != globalPass {
+		errText := "Неверный пароль"
+		err := jsonError(res, errText, http.StatusUnauthorized)
+		if err != nil {
+			logger.Println(err)
+		}
+		return
+	}
+	//Далее считаем, что юзер прошел аутентификацию
+	token, err := generateToken(creds.Password)
+	if err != nil {
+		errText := "Ошибка генерации токена"
+		logger.Printf("%s: %v", errText, err)
+		err = jsonError(res, errText, http.StatusInternalServerError)
+		if err != nil {
+			logger.Println(err)
+		}
+		return
+	}
+
+	cookie := &http.Cookie{
+		Name:     "token",
+		Value:    token,
+		Path:     "/",                     // Будет доступен по всему домену
+		HttpOnly: true,                    // Доступна только для браузера, JavaScript не видит
+		SameSite: http.SameSiteStrictMode, // Ограничивает доступ третьих сторон
+		Secure:   true,                    // Обязательная мера, если сайт работает по HTTPS
+		MaxAge:   24 * 60 * 60,            // Жизненный цикл куки — 8 часов
+	}
+	http.SetCookie(res, cookie)
+	//вроде поставил reddirect
+	// http.Redirect(res, req, "/", http.StatusSeeOther)
+	err = writeJson(res, token, http.StatusOK)
+	if err != nil {
+		logger.Println(err)
+	}
+}
+
+// Костыль, понимаю
+func Wrap(next func(w http.ResponseWriter, r *http.Request, logger *log.Logger), middle func(w http.ResponseWriter, r *http.Request, logger *log.Logger) bool, logger *log.Logger) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Сначала вызываем middleware
+		continueProcessing := middle(w, r, logger)
+
+		// Если middleware разрешает продолжение обработки
+		if continueProcessing {
+			// Вызываем основной обработчик
+			next(w, r, logger)
+		}
+	})
+}
+
+// func AuthMiddleware(next http.HandlerFunc) http.HandlerFunc
+
+//	func AuthMiddleware(next func(w http.ResponseWriter, r *http.Request, logger *log.Logger), logger *log.Logger) http.HandlerFunc {
+//		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+func AuthMiddleware(w http.ResponseWriter, r *http.Request, logger *log.Logger) bool {
+	var token *http.Cookie
+
+	pass := os.Getenv("TODO_PASSWORD")
+	// Если пароль не задан - пропускаем проверку
+	if len(pass) == 0 {
+		return true
+	}
+
+	token, err := r.Cookie("token")
+	if err == nil && isValidPassword(token.Value, pass) {
+		logger.Printf("Прошли прослойку аутентификации %s", r.URL)
+		return true
+	}
+	// дальнейшая логика работает если кука невалидна:
+	var errText string
+	var status int
+	switch {
+	case err != http.ErrNoCookie:
```

Comment:
Лучше так приучать себя делать 
```suggestion
	case !errors.Is(err, http.ErrNoCookie):
```